CvT: Introducing Convolutions to Vision Transformers

AnugunjNaman opened the pull request:
# What does this PR do?
Add CvT Model for Vision Classification

Fixes #13158


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@NielsRogge @FrancescoSaverioZuppichini

Commit:
Cvt Files

Commit:
Cvt Files

Commit:
Update labels for CvTConfig

Commit:
Update labels for CvTConfig

FrancescoSaverioZuppichini committed:
tracing

FrancescoSaverioZuppichini committed:
tracing

Commit:
added cls_token part

Commit:
added cls_token part

Commit:
added cls_token part

Commit:
Adding cls_part

Commit:
Updated Cvt

Commit:
Updated Cvt

Commit:
Error removed

Commit:
Error removed

AnugunjNaman committed:
Error removed

AnugunjNaman committed:
Error removed

AnugunjNaman committed:
CLS Error Removed

AnugunjNaman committed:
Error removed

AnugunjNaman committed:
Error removed

AnugunjNaman committed:
Update

AnugunjNaman committed:
Update

AnugunjNaman committed:
test for cvt

AnugunjNaman committed:
test for cvt

AnugunjNaman committed:
feature test

AnugunjNaman committed:
feature test

AnugunjNaman committed:
testings folder create

AnugunjNaman committed:
testings folder create

AnugunjNaman committed:
Update initialization

AnugunjNaman committed:
Update initialization

AnugunjNaman committed:
Tests passed

AnugunjNaman committed:
Update initialization

AnugunjNaman committed:
Update initialization

AnugunjNaman committed:
Doc Details - Cvt

AnugunjNaman committed:
Removed files not needed

AnugunjNaman committed:
Removed files not needed

FrancescoSaverioZuppichini committed:
removed cvt feature extractor, using convnext

FrancescoSaverioZuppichini committed:
huge refactor

FrancescoSaverioZuppichini committed:
removed einops

FrancescoSaverioZuppichini committed:
make style

AnugunjNaman committed:
Update rename weights scripts

AnugunjNaman committed:
Minor changes

AnugunjNaman committed:
Minor changes

AnugunjNaman committed:
Minor Changes

AnugunjNaman committed:
Minor Updates

AnugunjNaman committed:
cvt edit

AnugunjNaman committed:
Update

AnugunjNaman committed:
argparse conversion checkpoint cvt

AnugunjNaman committed:
Test cases error solved

AnugunjNaman committed:
Fixing styles

AnugunjNaman committed:
Fixing styles

FrancescoSaverioZuppichini reviewed (commented):
Thanks for working on this. Some comments, mostly about naming and code quality. 

We recently decide a naming convention, so:

- `hidden_state` indicates a tensor, usually of shape `(b, .....)`
- `hidden_states` indicates a list of tensors

Do not use contracted layers name, 

- `conv` -> `convolution`
- `norm` -> `normalization`
- etc

I advocate adding a `CVTStage` to each block simpler

AnugunjNaman commented:
Well I will need to update all weights again. Since they were in contracted form. Okay will take time.

AnugunjNaman commented on tests/cvt/test_modeling_cvt.py:
```diff
@@ -0,0 +1,352 @@
+        image_size=64,
+        num_channels=3,
+        num_stages=3,
+        embed_dim=[64, 192, 384],
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        patch_size = to_2tuple(patch_size)
+        self.patch_size = patch_size
+        self.projection = nn.Conv2d(num_channels, embed_dim, kernel_size=patch_size, stride=stride, padding=padding)
+        self.norm = nn.LayerNorm(embed_dim)
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        # rearrange "b c h w -> b (h w) c"
+        pixel_values = pixel_values.view(batch_size, num_channels, hidden_size).permute(0, 2, 1)
+        if self.norm:
+            pixel_values = self.norm(pixel_values)
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+class CvtSelfAttentionConvProjection(nn.Sequential):
+    def __init__(self, embed_dim, kernel_size, padding, stride):
+        super().__init__()
+        self.conv = nn.Conv2d(
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        return pixel_values
+
+
+class CvtSelfAttentionConvProjection(nn.Sequential):
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+
+
+class CvtSelfAttentionLinearProjection(nn.Module):
+    def __init__(self):
+        super().__init__()
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        return hidden_states
+
+
+class CvtSelfAttentionProjection(nn.Sequential):
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, hidden_states):
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        )
+        self.dropout = nn.Dropout(dropout_rate)
+
+    def forward(self, pixel_values):
+        embeddings = self.conv_embeddings(pixel_values)
+        embeddings = self.dropout(embeddings)
+        return embeddings
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        # rearrange 'b t (h d) -> b h t d'
+        return inputs.view(batch_size, hidden_size, self.num_heads, head_dim).permute(0, 2, 1, 3)
+
+    def forward(self, hidden_states, height, width, head_mask=None, output_attentions=False):
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        self.dense = nn.Linear(embed_dim, embed_dim)
+        self.dropout = nn.Dropout(drop_rate)
+
+    def forward(self, hidden_states, input_tensor):
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        super().__init__()
+        self.hidden_dim = int(embed_dim * mlp_ratio)
+        self.dense = nn.Linear(embed_dim, self.hidden_dim)
+        self.intermediate_act_fn = nn.GELU()
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+        self.dense = nn.Linear(self.hidden_dim, embed_dim)
+        self.dropout = nn.Dropout(drop_rate)
+
+    def forward(self, hidden_states, input_tensor):
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+            output = (logits,) + outputs[2:]
+            return ((loss,) + output) if loss is not None else output
+
+        return SequenceClassifierOutput(
```
Done

AnugunjNaman commented on src/transformers/models/cvt/convert_cvt_original_pytorch_checkpoit_to_pytorch.py:
```diff
@@ -0,0 +1,331 @@
+        config.depth = [2, 2, 20]
+        config.num_heads = [3, 12, 16]
+        config.embed_dim = [192, 768, 1024]
+
```
Done

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+            with_cls_token,
+        )
+
+        self.intermediate = CvtIntermediate(embed_dim, mlp_ratio)
```
I have no idea why intermediate is chosen. I followed the ViT and Segformer Convention. @NielsRogge can elaborate more here.

NielsRogge replied:
The term "intermediate" comes from the original BERT implementation, which was then also adopted by ViT.

It refers to the intermediate MLP after self-attention.

AnugunjNaman commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+
+        self.dropout = nn.Dropout(attention_drop_rate)
+
+    def rearrange_for_multi_head_attention(self, inputs):
```
Done

AnugunjNaman committed:
Update

AnugunjNaman committed:
Update Style

AnugunjNaman committed:
Update Style

AnugunjNaman committed:
Update Style

AnugunjNaman committed:
Update Style

AnugunjNaman commented:
@NielsRogge waiting for update!

NielsRogge commented on README.md:
```diff
@@ -248,7 +248,8 @@ Current number of checkpoints: ![](https://img.shields.io/endpoint?url=https://h
 1. **[CPM](https://huggingface.co/docs/transformers/model_doc/cpm)** (from Tsinghua University) released with the paper [CPM: A Large-scale Generative Chinese Pre-trained Language Model](https://arxiv.org/abs/2012.00413) by Zhengyan Zhang, Xu Han, Hao Zhou, Pei Ke, Yuxian Gu, Deming Ye, Yujia Qin, Yusheng Su, Haozhe Ji, Jian Guan, Fanchao Qi, Xiaozhi Wang, Yanan Zheng, Guoyang Zeng, Huanqi Cao, S ...
-1. **[CTRL](https://huggingface.co/docs/transformers/model_doc/ctrl)** (from Salesforce) released with the paper [CTRL: A Conditional Transformer Language Model for Controllable Generation](https://arxiv.org/abs/1909.05858) by Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher.
+1. **[CTRL](https://huggingface.co/docs/transformers/model_doc/ctrl)** (from Salesforce) released with the paper [CTRL: A Conditional Transformer Language Model for Controllable Generation](https://arxiv.org/abs/2103.15808) by Haiping Wu, Bin Xiao, Noel Codella, Mengchen Liu, Xiyang Dai, Lu Yuan and Lei Zhang.
+1. **[CVT](https://huggingface.co/docs/transformers/model_doc/cvt)** (from Microsoft) released with the paper [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/1909.05858) by Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher.
```
```suggestion
1. **[CvT](https://huggingface.co/docs/transformers/model_doc/cvt)** (from Microsoft) released with the paper [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/1909.05858) by Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher.
```

NielsRogge commented on docs/source/index.mdx:
```diff
@@ -186,6 +187,7 @@ Flax), PyTorch, and/or TensorFlow.
 |          ConvBERT           |       ✅       |       ✅       |       ✅        |         ✅         |      ❌      |
 |          ConvNext           |       ❌       |       ❌       |       ✅        |         ✅         |      ❌      |
 |            CTRL             |       ✅       |       ❌       |       ✅        |         ✅         |      ❌      |
+|             Cvt             |       ❌       |       ❌       |       ✅        |         ❌         |      ❌      |
```
```suggestion
|             CvT             |       ❌       |       ❌       |       ✅        |         ❌         |      ❌      |
```

NielsRogge commented on docs/source/model_doc/cvt.mdx:
```diff
@@ -0,0 +1,59 @@
+specific language governing permissions and limitations under the License.
+-->
+
+# Convolutional Vision Transformer (Cvt)
```
```suggestion
# Convolutional Vision Transformer (CvT)

```

NielsRogge commented on docs/source/model_doc/cvt.mdx:
```diff
@@ -0,0 +1,59 @@
+
+## Overview
+
+The Cvt model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808)  by 
```
```suggestion
The CvT model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808)  by 

```

NielsRogge commented on docs/source/model_doc/cvt.mdx:
```diff
@@ -0,0 +1,59 @@
+## Overview
+
+The Cvt model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808)  by 
+Haiping Wu, Bin Xiao, Noel Codella, Mengchen Liu, Xiyang Dai, Lu Yuan and Lei Zhang. Convolutional vision Transformer (CvT),
+that improves [Vision Transformer (ViT)](https://arxiv.org/abs/2010.11929) in performance and efficiency by introducing convolutions into ViT to yield the best of both designs.
```
```suggestion
Haiping Wu, Bin Xiao, Noel Codella, Mengchen Liu, Xiyang Dai, Lu Yuan and Lei Zhang. The Convolutional vision Transformer (CvT) improves the [Vision Transformer (ViT)](vit) in performance and efficiency by introducing convolutions into ViT to yield the best of both designs.

```
Here we can link to the vit docs.

NielsRogge commented on docs/source/model_doc/cvt.mdx:
```diff
@@ -0,0 +1,59 @@
+
+Tips:
+
+
```
```suggestion
```

NielsRogge commented on docs/source/model_doc/cvt.mdx:
```diff
@@ -0,0 +1,59 @@
+Tips:
+
+
+- Cvt models are regular Vision Transformers, but trained with convolutions. It outperform the [original model (ViT)](vit) 
```
```suggestion
- CvT models are regular Vision Transformers, but trained with convolutions. They outperform the [original model (ViT)](vit) 

```

NielsRogge commented on src/transformers/__init__.py:
```diff
@@ -115,9 +115,10 @@
         "load_tf2_weights_in_pytorch_model",
     ],
-    "models": [],
     # Models
+    "models": [],
     "models.albert": ["ALBERT_PRETRAINED_CONFIG_ARCHIVE_MAP", "AlbertConfig"],
+    # Models
```
Something weird happened here; normally this PR shouldn't change this piece of code.

AnugunjNaman replied:
No clue about this one

NielsRogge replied:
You can just look at how it looks in the main branch, and then update the code here.

NielsRogge commented on src/transformers/__init__.py:
```diff
@@ -639,6 +641,7 @@
     _import_structure["modeling_utils"] = ["Conv1D", "PreTrainedModel", "apply_chunking_to_forward", "prune_layer"]
 
     # PyTorch models structure
+
```
```suggestion
```

NielsRogge commented on src/transformers/__init__.py:
```diff
@@ -2897,6 +2909,7 @@
         from .utils.dummy_scatter_objects import *
 
     if is_torch_available():
+
```
```suggestion
```

NielsRogge commented on src/transformers/models/auto/configuration_auto.py:
```diff
@@ -35,6 +35,7 @@
         ("maskformer", "MaskFormerConfig"),
         ("decision_transformer", "DecisionTransformerConfig"),
         ("poolformer", "PoolFormerConfig"),
+        ("cvt", "CvtConfig"),
```
Sorry to disappoint you, but the "v" comes after "o", so you can place "cvt" after "convnext" everywhere ;)

AnugunjNaman replied:
Done

NielsRogge commented on src/transformers/models/cvt/configuration_cvt.py:
```diff
@@ -0,0 +1,161 @@
+logger = logging.get_logger(__name__)
+
+CVT_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "msft/cvt": "https://huggingface.co/msft/cvt/resolve/main/config.json",
```
```suggestion
    "microsoft/cvt": "https://huggingface.co/microsoft/cvt/resolve/main/config.json",
```
All checkpoints should be placed under the "microsoft" organization (I can help with that).

NielsRogge commented on src/transformers/models/cvt/configuration_cvt.py:
```diff
@@ -0,0 +1,161 @@
+
+class CvtConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`~CvtModel`]. It is used to instantiate an Cvt
```
```suggestion
    This is the configuration class to store the configuration of a [`CvtModel`]. It is used to instantiate an Cvt
```

NielsRogge commented on src/transformers/models/cvt/configuration_cvt.py:
```diff
@@ -0,0 +1,161 @@
+    r"""
+    This is the configuration class to store the configuration of a [`~CvtModel`]. It is used to instantiate an Cvt
+    model according to the specified arguments, defining the model architecture. Instantiating a configuration with the
+    defaults will yield a similar configuration to that of the Cvt [msft/cvt](https://huggingface.co/msft/cvt)
+    architecture.
```
```suggestion
    defaults will yield a similar configuration to that of the Cvt [microsoft/cvt](https://huggingface.co/microsoft/cvt)
    architecture.
```

NielsRogge commented on tests/cvt/test_modeling_cvt.py:
```diff
@@ -0,0 +1,281 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team. All rights reserved.
```
```suggestion
# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
```

NielsRogge commented on utils/get_modified_files.py:
```diff
@@ -24,7 +24,7 @@
 
 
-fork_point_sha = subprocess.check_output("git merge-base main HEAD".split()).decode("utf-8")
+fork_point_sha = subprocess.check_output("git merge-base master HEAD".split()).decode("utf-8")
```
This file should not be changed (as we renamed master to main).

AnugunjNaman replied:
Done

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+# coding=utf-8
+# Copyright 2021 Google AI, Ross Wightman, The HuggingFace Inc. team. All rights reserved.
```
```suggestion
# Copyright 2022 Microsoft Research and The HuggingFace Inc. team. All rights reserved.
```

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch Cvt model."""
```
```suggestion
""" PyTorch CvT model."""
```

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+            shape `(batch_size, sequence_length, hidden_size)`. Hidden-states of the model at the output of each layer
+            plus the initial embedding outputs.
+        cls_token (`torch.FloatTensor` of shape `(batch_size, 1, hidden_size)`):
+            Classification Token at the output of the last layer of the model.
```
```suggestion
            Classification token at the output of the last layer of the model.
```

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+    return _no_grad_trunc_normal_(tensor, mean, std, a, b)
+
+
+class DropPath(nn.Module):
```
```suggestion
class CvtDropPath(nn.Module):
```

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+
+
+class CvtEmbeddings(nn.Module):
+    """
+    Construct the CLS token, position and patch embeddings.
+
+    """
```
I don't think this docstring is correct

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+        self.dropout = nn.Dropout(attention_drop_rate)
+
+    def rearrange_for_multi_head_attention(self, hidden_state):
+        (
+            batch_size,
+            hidden_size,
+            _,
+        ) = hidden_state.shape
```
Can fit in one line

AnugunjNaman replied:
Done

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.bias is not None:
+                module.bias.data.zero_()
+        elif isinstance(module, nn.Embedding):
+            module.weight.data.normal_(mean=0.0, std=self.config.initializer_range)
+            if module.padding_idx is not None:
+                module.weight.data[module.padding_idx].zero_()
```
Is there an `nn.Embedding` class in this model? If not, then this can be removed.

AnugunjNaman replied:
Done

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+_IMAGE_CLASS_EXPECTED_OUTPUT = "'tabby cat'"
+
+
+CVT_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "anugunj/cvt-13",
+    "anugunj/cvt-13-384-1k",
+    "anugunj/cvt-13-384-22k",
+    "anugunj/cvt-21",
+    "anugunj/cvt-21-384-1k",
+    "anugunj/cvt-21-384-22k",
+    # See all Cvt models at https://huggingface.co/models?filter=cvt
+]
```
To be updated

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+
+
+@dataclass
+class BaseModelOutputWithCLSToken(ModelOutput):
```
Wondering if the cls token can't be added in any case as the first `last_hidden_state`?

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+            Classification Token at the output of the last layer of the model.
+    """
+
+    last_hidden_state: torch.FloatTensor = None
+    cls_token_value: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
```
The names of the attributes here aren't the same as the docstring.

AnugunjNaman replied:
Done

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+
+
+# Inspired by
+# https://github.com/rwightman/pytorch-image-models/blob/b9bd960a032c75ca6b808ddeed76bee5f3ed4972/timm/models/layers/helpers.py
+# From PyTorch internals
```
```suggestion
# Copied from transformers.models.vit.modeling_vit.to_2tuple
```

You can replace this with a copied from statement

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+    return x * random_tensor
+
+
+def _no_grad_trunc_normal_(tensor, mean, std, a, b):
```
I don't see this function being used anywhere?

AnugunjNaman replied:
Done

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+        return tensor
+
+
+def trunc_normal_(tensor, mean=0.0, std=1.0, a=-2.0, b=2.0):
```
If the function above can be removed, then this one as well.

AnugunjNaman replied:
Done

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,885 @@
+    """
+    def __init__(self, patch_size, num_channels, embed_dim, stride, padding):
+        super().__init__()
+        patch_size = to_2tuple(patch_size)
```
For now, I haven't removed it in other implementations.

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+
+class CvtLayer(nn.Module):
+    """
+    CvtLayer composed by attention layers, normalization and mlps
```
```suggestion
    CvtLayer composed by attention layers, normalization and multi-layer perceptrons.
```

NielsRogge reviewed (commented):
Looks good already! A few comments:

- wondering if there's a need for `BaseModelOutputWithCLSToken` => can we by default add the CLS token? Users can always do `outputs.last_hidden_state[:,1,:]` if they don't want it.
- is there a need for a `CvtFeatureExtractor`? What are the minimal image transformations required for inference?

AnugunjNaman commented:
I will be free on weekend, I make the suggested changes then 

1. I think it’s okay, more in line with repo and paper. More clear I guess.
2. I talked with Francesco, he said no need for that since it’s similar to ConvNextFeatureExteactor use AutoFeatureExtractor.


I was hoping to have a meeting were we can quickly finish this. As you suggested changes are minor. Maybe an hour or two it can be done.

NielsRogge commented on src/transformers/models/cvt/modeling_cvt.py:
```diff
@@ -0,0 +1,828 @@
+
+# General docstring
+_CONFIG_FOR_DOC = "CvtConfig"
+_FEAT_EXTRACTOR_FOR_DOC = "CvtFeatureExtractor"
```
```suggestion
_FEAT_EXTRACTOR_FOR_DOC = "AutoFeatureExtractor"
```

NielsRogge commented:
> I was hoping to have a meeting were we can quickly finish this. As you suggested changes are minor. Maybe an hour or two it can be done. If your free any day this week we can do that then too.

Is it possible for you to apply the suggestions from the review?

AnugunjNaman committed:
Update README.md

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/index.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/__init__.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/__init__.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/configuration_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update tests/cvt/test_modeling_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update tests/cvt/test_modeling_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/modeling_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/modeling_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/modeling_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/modeling_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/modeling_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/modeling_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/configuration_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update src/transformers/models/cvt/configuration_cvt.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

NielsRogge commented on docs/source/model_doc/cvt.mdx:
```diff
@@ -15,7 +15,8 @@ specific language governing permissions and limitations under the License.
 ## Overview
 
-The Cvt model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808)  by 
+The CvT model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808)  by 
```
```suggestion
The CvT model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808) by 

```

NielsRogge commented on docs/source/model_doc/cvt.mdx:
```diff
@@ -15,7 +15,8 @@ specific language governing permissions and limitations under the License.
+The CvT model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808)  by 
+
 Haiping Wu, Bin Xiao, Noel Codella, Mengchen Liu, Xiyang Dai, Lu Yuan and Lei Zhang. Convolutional vision Transformer (CvT),
 that improves [Vision Transformer (ViT)](https://arxiv.org/abs/2010.11929) in performance and efficiency by introducing convolutions into ViT to yield the best of both designs.
```
```suggestion
The CvT model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808) by Haiping Wu, Bin Xiao, Noel Codella, Mengchen Liu, Xiyang Dai, Lu Yuan and Lei Zhang. Convolutional vision Transformer (CvT),
that improves [Vision Transformer (ViT)](https://arxiv.org/abs/2010.11929) in performance and efficiency by introducing convolutions into ViT to yield the best of both designs.

```

AnugunjNaman committed:
Updates

AnugunjNaman committed:
Update with fixup

AnugunjNaman committed:
Update with fixup

AnugunjNaman commented:
@NielsRogge I made the suggested changes. The shifting of models to Microsoft is left. You can review it.

NielsRogge commented on README_ko.md:
```diff
@@ -230,6 +230,8 @@ Flax, PyTorch, TensorFlow 설치 페이지에서 이들을 conda로 설치하는
 1. **[ConvNeXT](https://huggingface.co/docs/transformers/main/model_doc/convnext)** (from Facebook AI) released with the paper [A ConvNet for the 2020s](https://arxiv.org/abs/2201.03545) by Zhuang Liu, Hanzi Mao, Chao-Yuan Wu, Christoph Feichtenhofer, Trevor Darrell, Saining Xie.
 1. **[CPM](https://huggingface.co/docs/transformers/model_doc/cpm)** (from Tsinghua University) released with the paper [CPM: A Large-scale Generative Chinese Pre-trained Language Model](https://arxiv.org/abs/2012.00413) by Zhengyan Zhang, Xu Han, Hao Zhou, Pei Ke, Yuxian Gu, Deming Ye, Yujia Qin, Yusheng Su, Haozhe Ji, Jian Guan, Fanchao Qi, Xiaozhi Wang, Yanan Zheng, Guoyang Zeng, Huanqi Cao, S ...
 1. **[CTRL](https://huggingface.co/docs/transformers/model_doc/ctrl)** (from Salesforce) released with the paper [CTRL: A Conditional Transformer Language Model for Controllable Generation](https://arxiv.org/abs/1909.05858) by Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher.
+1. **[CVT](https://huggingface.co/docs/transformers/model_doc/cvt)** (from Microsoft) released with the paper [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/1909.05858) by Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher.
+1. **[CvT](https://huggingface.co/docs/transformers/main/model_doc/cvt)** (from Microsoft) released with the paper [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/1909.05858) by Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher.
```
This seems like a mistake to me, CvT is included twice here. Running `make fix-copies` will fix this issue.

NielsRogge commented on README_zh-hans.md:
```diff
@@ -254,6 +254,8 @@ conda install -c huggingface transformers
 1. **[ConvNeXT](https://huggingface.co/docs/transformers/main/model_doc/convnext)** (来自 Facebook AI) 伴随论文 [A ConvNet for the 2020s](https://arxiv.org/abs/2201.03545) 由 Zhuang Liu, Hanzi Mao, Chao-Yuan Wu, Christoph Feichtenhofer, Trevor Darrell, Saining Xie 发布。
 1. **[CPM](https://huggingface.co/docs/transformers/model_doc/cpm)** (来自 Tsinghua University) 伴随论文 [CPM: A Large-scale Generative Chinese Pre-trained Language Model](https://arxiv.org/abs/2012.00413) 由 Zhengyan Zhang, Xu Han, Hao Zhou, Pei Ke, Yuxian Gu, Deming Ye, Yujia Qin, Yusheng Su, Haozhe Ji, Jian Guan, Fanchao Qi, Xiaozhi Wang, Yanan Zheng, Guoyang Zeng, Huanqi Cao, Shengqi Chen, Daixuan L ...
 1. **[CTRL](https://huggingface.co/docs/transformers/model_doc/ctrl)** (来自 Salesforce) 伴随论文 [CTRL: A Conditional Transformer Language Model for Controllable Generation](https://arxiv.org/abs/1909.05858) 由 Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher 发布。
+1. **[CVT](https://huggingface.co/docs/transformers/model_doc/cvt)** (来自 Microsoft) 伴随论文 [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/1909.05858) 由 Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher 发布。
+1. **[CvT](https://huggingface.co/docs/transformers/main/model_doc/cvt)** (来自 Microsoft) 伴随论文 [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/1909.05858) 由 Nitish Shirish Keskar*, Bryan McCann*, Lav R. Varshney, Caiming Xiong and Richard Socher 发布。
```
Same here

NielsRogge commented on docs/source/en/index.mdx:
```diff
@@ -188,6 +189,7 @@ Flax), PyTorch, and/or TensorFlow.
 |          ConvBERT           |       ✅       |       ✅       |       ✅        |         ✅         |      ❌      |
 |          ConvNext           |       ❌       |       ❌       |       ✅        |         ✅         |      ❌      |
 |            CTRL             |       ✅       |       ❌       |       ✅        |         ✅         |      ❌      |
+|             Cvt             |       ❌       |       ❌       |       ✅        |         ❌         |      ❌      |
```
```suggestion
|             CvT             |       ❌       |       ❌       |       ✅        |         ❌         |      ❌      |
```

NielsRogge commented on docs/source/en/model_doc/cvt.mdx:
```diff
@@ -0,0 +1,62 @@
+
+## Overview
+
+The CvT model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808)  by 
+
+Haiping Wu, Bin Xiao, Noel Codella, Mengchen Liu, Xiyang Dai, Lu Yuan and Lei Zhang. The Convolutional vision Transformer (CvT) improves the [Vision Transformer (ViT)](vit) in performance and efficiency by introducing convolutions into ViT to yield the best of both designs.
```
```suggestion
The CvT model was proposed in [CvT: Introducing Convolutions to Vision Transformers](https://arxiv.org/abs/2103.15808) by Haiping Wu, Bin Xiao, Noel Codella, Mengchen Liu, Xiyang Dai, Lu Yuan and Lei Zhang. The Convolutional vision Transformer (CvT) improves the [Vision Transformer (ViT)](vit) in performance and efficiency by introducing convolutions into ViT to yield the best of both designs.

```

NielsRogge commented on docs/source/en/model_doc/cvt.mdx:
```diff
@@ -0,0 +1,62 @@
+
+Tips:
+
+- CvT models are regular Vision Transformers, but trained with convolutions. They outperform the [original model (ViT)](vit) 
+
+  when fine-tuned on ImageNet-1K and CIFAR-100. You can check out demo notebooks regarding inference as well as
+  fine-tuning on custom data [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/VisionTransformer) (you can just replace [`ViTFeatureExtractor`] by [`AutoFeatureExtractor`] and [`ViTForImageClassification`] by [`CvtForImageClassification`]).
+
+- The available checkpoints are either (1) pre-trained on [ImageNet-22k](http://www.image-net.org/) (a collection of
```
```suggestion
- CvT models are regular Vision Transformers, but trained with convolutions. They outperform the [original model (ViT)](vit) when fine-tuned on ImageNet-1K and CIFAR-100. You can check out demo notebooks regarding inference as well as fine-tuning on custom data [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/VisionTransformer) (you can just replace [`ViTFeatureExtractor`] by [`AutoFeatureExtractor`] and [`ViTForImageClassification`] by [`CvtForImageClassification`]).
- The available checkpoints are either (1) pre-trained on [ImageNet-22k](http://www.image-net.org/) (a collection of

```

NielsRogge commented on docs/source/en/model_doc/cvt.mdx:
```diff
@@ -0,0 +1,62 @@
+  14 million images and 22k classes) only, (2) also fine-tuned on ImageNet-22k or (3) also fine-tuned on [ImageNet-1k](http://www.image-net.org/challenges/LSVRC/2012/) (also referred to as ILSVRC 2012, a collection of 1.3 million
+  images and 1,000 classes).
+
+
```
```suggestion
```

NielsRogge commented on src/transformers/models/auto/configuration_auto.py:
```diff
@@ -236,6 +238,7 @@
         ("maskformer", "MaskFormer"),
         ("poolformer", "PoolFormer"),
         ("convnext", "ConvNext"),
+        ("cvt", "Cvt"),
```
```suggestion
        ("cvt", "CvT"),
```
You can run `make fix-copies` again after this.

NielsRogge commented on src/transformers/models/cvt/configuration_cvt.py:
```diff
@@ -0,0 +1,160 @@
+            The size (resolution) of each image.
+        num_channels (`int`, *optional*, defaults to 3):
+            The number of input channels.
+        num_stages (`int`, *optional*, defaults to 3):
```
Yes I agree, the num_stages argument can be removed, and instead you can leverage `len(depth)`

NielsRogge replied:
It can also be beneficial to check at initialization of the config that patch_sizes, patch_stride etc. all have the same length

NielsRogge commented on src/transformers/models/cvt/configuration_cvt.py:
```diff
@@ -0,0 +1,160 @@
+            The bool for position embeddings
+        cls_token (`List[bool]`, *optional*, defaults to [False, False, True]]):
+            The bool for cls_token
+        qkv_projection_method (`List[string]`, *optional*, defaults to 'dw_bn', 'dw_bn', 'dw_bn']]):
```
Yes, dw_bn for instance is probably depth-wise batch normalization.

AnugunjNaman replied:
This is done in explaining.

NielsRogge commented on src/transformers/models/cvt/convert_cvt_original_pytorch_checkpoit_to_pytorch.py:
```diff
@@ -0,0 +1,352 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
```
There's a mistake in the name of this file, checkpoit -> checkpoint.

NielsRogge commented:
Some remarks:

* PR is in a good state, but please don't close comments that aren't resolved yet. I would like the BaseModelOutputWithCLSToken to be removed, and just return the CLS token by default.
* can you also add CvT to the doc tests? This will make sure all code examples in the docstrings are tested as well. All details found here: https://github.com/huggingface/transformers/tree/main/docs#testing-documentation-examples

AnugunjNaman committed:
Update docs/source/en/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/en/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman committed:
Update docs/source/en/model_doc/cvt.mdx

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

AnugunjNaman commented:
@NielsRogge Can you work on the doc part. Every time something or the other mistake takes place. If the doc part is done. 

Then I can make changes in `BaseModelOutputCLSToken` and inference of stages using depth and length check on patch sizes etc.

sgugger committed:
Fix batch size in evaluation loop (#16763)

* Fix batch size in evaluation loop

* remove debug statement

muellerzr committed:
Make nightly install dev accelerate (#16783)

stas00 committed:
[deepspeed / m2m_100] make deepspeed zero-3 work with layerdrop (#16717)

* [deepspeed / m2m_100] make deepspeed 3 work with layerdrop

* fix

* revert last

sgugger committed:
Kill async pushes when calling push_to_hub with blocking=True (#16755)

NielsRogge committed:
Improve image classification example (#16585)

* Improve README

* Make dataset_name argument optional

* Improve local data

* Fix bug

* Improve README some more

* Apply suggestions from code review

* Improve README

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

sanchit-gandhi committed:
[SpeechEncoderDecoderModel] Fix bug in reshaping labels (#16748)

code-review-doctor committed:
Fix issue avoid-missing-comma found at https://codereview.doctor (#16768)

stas00 committed:
[trainer / deepspeed] fix hyperparameter_search (#16740)

* [trainer / deepspeed] fix hyperparameter_search

* require optuna

* style

* oops

* add dep in the right place

* create deepspeed-testing dep group

* Trigger CI

stas00 committed:
[modeling utils] revamp `from_pretrained(..., low_cpu_mem_usage=True)` + tests (#16657)

* add low_cpu_mem_usage tests

* wip: revamping

* wip

* install /usr/bin/time

* wip

* cleanup

* cleanup

* cleanup

* cleanup

* cleanup

* fix assert

* put the wrapper back

* cleanup; switch to bert-base-cased

* Trigger CI

* Trigger CI

ydshieh committed:
Fix PT TF ViTMAE (#16766)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

NielsRogge committed:
Update README.md (#16797)

sgugger committed:
Pin Jax to last working release (#16808)

* Pin Jax to last working release

* Try lower

* Try lower

gante committed:
CI: non-remote GH Actions now use a python venv (#16789)

gante committed:
TF generate refactor - XLA sample (#16713)

allanj committed:
Raise error and suggestion when using custom optimizer with Fairscale or Deepspeed (#16786)

* optimizer issues related to saving

* remove the "optimizer saving" option

* reformat using make style

gante committed:
Create empty venv on cache miss (#16816)

sgugger committed:
Fix syntax error in TorchHub workflow

NielsRogge committed:
[ViT, BEiT, DeiT, DPT] Improve code (#16799)

* Improve code

* Fix bugs

* Fix another bug

* Clean up DTP as well

* Update DPT model outputs

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

patrickvonplaten committed:
[Quicktour Audio] Improve && remove ffmpeg dependency (#16723)

* [Quicktour Audio] Improve && remove ffmpeg dependency

* final fix

* final touches

itszyin committed:
fix megatron bert convert state dict naming (#15820)

patrickvonplaten committed:
[Data2Vec] Add data2vec vision (#16760)

* save intermediate

* add vision

* add vision

* save

* finish models

* finish models

* continue

* finish

* up

* up

* up

* tests all pass

* clean up

* up

* up

* fix bugs in beit

* correct docs

* finish

* finish docs

* make style

* up

* more fixes

* fix type hint

* make style

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update tests/data2vec/test_modeling_data2vec_vision.py

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

* fix test

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

nbroad1881 committed:
use base_version to check torch version in torch_less_than_1_11 (#16806)

* use base_version

* make is_torch_less_than_1_8 match 1_11

Co-authored-by: Nicholas Broad <nicholas@nmbroad.com>

jsnfly committed:
Allow passing encoder_ouputs as tuple to EncoderDecoder Models (#16814)

* Add passing encoder_outputs as tuple to existing test

* Add check for tuple

* Add check for tuple also for speech and vision

Co-authored-by: jsnfly <jsnfly@gmx.de>

LysandreJik committed:
Refactor issues with yaml (#16772)

* Refactor issues with yaml

* Update .github/ISSUE_TEMPLATE/bug-report.yml

Co-authored-by: Anton Lozhkov <aglozhkov@gmail.com>

* Update .github/ISSUE_TEMPLATE/bug-report.yml

Co-authored-by: Anton Lozhkov <aglozhkov@gmail.com>

* Update .github/ISSUE_TEMPLATE/feature-request.yml

Co-authored-by: Sanchit Gandhi <93869735+sanchit-gandhi@users.noreply.github.com>

* Update .github/ISSUE_TEMPLATE/bug-report.yml

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

* Update .github/ISSUE_TEMPLATE/bug-report.yml

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

* Address review comments

Co-authored-by: Anton Lozhkov <aglozhkov@gmail.com>
Co-authored-by: Sanchit Gandhi <93869735+sanchit-gandhi@users.noreply.github.com>
Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

dlwh committed:
fix _setup_devices in case where there is no torch.distributed package in build (#16821)

* fix _setup_devices in case where there is not torch.distributed

* in training_args_sm.py as well

NielsRogge committed:
Clean up semantic segmentation tests (#16801)

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

NielsRogge committed:
Add semantic script no trainer, v2 (#16788)

* Add first draft from previous PR

* First draft

* Improve README and remove num_labels

* Make script more aligned with other scripts

* Improve README and apply suggestion from code review

qqaatw committed:
Fix `LayoutLMv2` tokenization docstrings (#16187)

* Fix docstrings

* Fix up

* Fix

ArthurZucker committed:
Wav2 vec2 phoneme ctc tokenizer optimisation (#16817)

* Solved href rendering issue in heading

Markdown references in headings such as '####' don't render well.
Replaced it with <h4>...<a></a></h> banners.

* PhonemeTokenizer optimization using phonemizer lib

The backend should only be initialized once, otherwise it is reloaded.
Added `init_backend` function, intializes a backend attribute.
Phonemize re-uses self.backend.
Should give ~10 times faster phonemization.

* formatted file with make style

* Documentation suggestion

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update /tokenization_wav2vec2_phoneme.py based on PR suggestion

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update CONTRIBUTING.md

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

patil-suraj committed:
[Flax] improve large model init and loading  (#16148)

* begin do_init

* add params_shape_tree

* raise error if params are accessed when do_init is False

* don't allow do_init=False when keys are missing

* make shape tree a property

* assign self._params at the end

* add test for do_init

* add do_init arg to all flax models

* fix param setting

* disbale do_init for composite models

* update test

* add do_init in FlaxBigBirdForMultipleChoice

* better names and errors

* improve test

* style

* add a warning when do_init=False

* remove extra if

* set params after _required_params

* add test for from_pretrained

* do_init => _do_init

* chage warning to info

* fix typo

* add params in init_weights

* add params to gpt neo init

* add params to init_weights

* update do_init test

* Trigger CI

* Apply suggestions from code review

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

* update template

* trigger CI

* style

* style

* fix template

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

code-review-doctor committed:
Some tests misusing assertTrue for comparisons fix (#16771)

* Fix issue avoid-misusing-assert-true found at https://codereview.doctor

* fix tests

* fix tf

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

Dahlbomii committed:
Type hints added for TFMobileBert (#16505)

* Type hints added

* make style

* Return type hints added

* fixed typo

Co-authored-by: matt <rocketknight1@gmail.com>

dandelin committed:
fix `rum_clm.py` seeking text column name twice (#16624)

echarlaix committed:
Add onnx export of models with a multiple choice classification head (#16758)

* Add export of models with a multiple-choice classification head

patrickvonplaten committed:
[ASR Pipeline] Correct init docs (#16833)

* correct

* up

NielsRogge committed:
Add image classification script, no trainer (#16727)

* Add first draft

* Improve README and run fixup

* Make script aligned with other scripts, improve README

* Improve script and add test

* Remove print statement

* Apply suggestions from code review

* Add num_labels to make test pass

* Improve README

wiio12 committed:
Add doc about `attention_mask` on gpt2  (#16829)

* Add doc about `attention_mask` on gpt2

Add a simple sentence describing how `attention_mask` needs to be constructed when ``past_key_values` is used.

* Add doc about attention_mask on gpt2_tf

* clean up style

* remove empty line white spaces

* remove whitespace in empty line

gante committed:
TF: Add sigmoid activation function (#16819)

Commit:
Correct Logging of Eval metric to Tensorboard (#16825)

* Correct Logging of Eval metric to Tensorboard

An empty dictionary ``eval_metrics`` was being logged, is replaced by ``eval_metric`` which is the output dictionary of ``metric.compute()``.

* Remove unused variable

SaulLu committed:
replace `Speech2TextTokenizer` by `Speech2TextFeatureExtractor` in some docstrings (#16835)

* replace `Speech2TextTokenizer` by `Speech2TextFeatureExtractor` in docstring

* quality

Dahlbomii committed:
Type hints added to Speech to Text (#16506)

* Type hints added

* return hints added

* Update src/transformers/models/speech_to_text/modeling_tf_speech_to_text.py

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

ydshieh committed:
Improve test_pt_tf_model_equivalence on PT side (#16731)

* Update test_pt_tf_model_equivalence on PT side

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

manuelciosici committed:
Add support for bitsandbytes (#15622)

* Add initial BNB integration

* fixup! Add initial BNB integration

* Add bnb test decorator

* Update Adamw8bit option name

* Use the full bnb package name

* Overide bnb for all embedding layers

* Fix package name

* Formatting

* Remove unnecessary import

* Update src/transformers/trainer.py

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>

* Rename AdamwBNB optimizer option

* Add training test checking that bnb memory utilization is lower

* fix merge

* fix merge; fix + extend new test

* cleanup

* expand bnb

* move all require_* candidates to testing_utils.py

Co-authored-by: Stas Bekman <stas00@users.noreply.github.com>
Co-authored-by: Stas Bekman <stas@stason.org>

patrickvonplaten committed:
[Typo] Fix typo in modeling utils (#16840)

mingboiz committed:
add DebertaV2 fast tokenizer (#15529)

Co-authored-by: alcinos <carion.nicolas@gmail.com>
Co-authored-by: SaulLu <55560583+SaulLu@users.noreply.github.com>
Co-authored-by: Nicolas Carion <carion.nicolas@gmail.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Narsil committed:
Fixing return type tensor with `num_return_sequences>1`. (#16828)

* Fixing return type tensor with `num_return_sequences>1`.

* Nit.

stas00 committed:
[modeling_utils] use less cpu memory with sharded checkpoint loading (#16844)

* less cpu memory with sharded checkpoint loading

* Trigger CI

* Trigger CI

stas00 committed:
[docs] fix url (#16860)

sgugger committed:
Fix custom init sorting script (#16864)

muellerzr committed:
Fix multiproc metrics in no_trainer examples (#16865)

Narsil committed:
Long QuestionAnsweringPipeline fix. (#16778)

* Temporary commit witht the long QA fix.

* Adding slow tests covering this fix.

* Removing fast test as it doesn't fail anyway.

stefan-it committed:
t5: add conversion script for T5X to FLAX (#16853)

* t5: add conversion script for T5X to FLAX

* t5: make flake happy

* t5: add copyright message to t5x conversion script

* t5: fix lm head for v1.0 checkpoints

garyhlai committed:
tiny tweak to allow BatchEncoding.token_to_char when token doesn't correspond to chars (#15901)

* tweak to allow BatchEncoding.char_to_token(0)

* update docstring

* remote trailing whitespace

* make fixup

* make value checking for span_indices explicit

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Narsil committed:
Adding support for `array` key in raw dictionnaries in ASR pipeline. (#16827)

* Adding support for `array` key in raw dictionnaries in ASR pipeline.

* ES .

* Update src/transformers/pipelines/automatic_speech_recognition.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Making it work by not popping `array` first.

* Black 22.3

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

sgugger committed:
Return input_ids in ImageGPT feature extractor (#16872)

eldarkurtic committed:
Use ACT2FN to fetch ReLU activation (#16874)

- all activations should be fetched through ACT2FN
- it returns ReLU as `nn.Module`, which allows attaching hooks on the activation function and prints it to stdout when `print(model)`

chainyo committed:
Fix GPT-J onnx conversion (#16780)

* add gptj to TOKENIZER_MAPPING_NAMES

* fix int32 to float to avoid problem in onnx

* Update src/transformers/models/gptj/modeling_gptj.py

Co-authored-by: ChainYo <t.chaigneau.tc@gmail.com>
Co-authored-by: lewtun <lewis.c.tunstall@gmail.com>

ydshieh committed:
Fix doctest list (#16878)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

loubnabnl committed:
New features for CodeParrot training script (#16851)

* add tflops logging and fix grad accumulation

* add accelerate tracking and checkpointing

* scale loss of last batch correctly

* fix typo

* compress loss computation

Co-authored-by: Leandro von Werra <lvwerra@users.noreply.github.com>

* add resume from checkpoint argument

* add load_state accelerate from checkpoint, register lr scheduler and add tflops function

* reformat code

* reformat code

* add condition on path for resume checkpoint

* combine if conditions

Co-authored-by: Leandro von Werra <lvwerra@users.noreply.github.com>

* add source for tflops formula

Co-authored-by: Leandro von Werra <lvwerra@users.noreply.github.com>

ydshieh committed:
Add missing entries in mappings (#16857)

* add missing entries in some mappings

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

gante committed:
TF: rework XLA generate tests (#16866)

mariosasko committed:
Minor fixes/improvements in `convert_file_size_to_int` (#16891)

* Minor improvements to `convert_file_size_to_int`

* Add <unit>bit version to kilos and megas

* Minor fix

vumichien committed:
Add doc tests for Albert and Bigbird (#16774)

* Add doctest BERT

* make fixup

* fix typo

* change checkpoints

* make fixup

* define doctest output value, update doctest for mobilebert

* solve fix-copies

* update QA target start index and end index

* change checkpoint for docs and reuse defined variable

* Update src/transformers/models/bert/modeling_tf_bert.py

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

* make fixup

* Add Doctest for Albert and Bigbird

* make fixup

* overwrite examples for Albert and Bigbird

* Apply suggestions from code review

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

* update longer examples for Bigbird

* using examples from squad_v2

* print out example text

* change name token-classification-big-bird checkpoint to random

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>
Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

chainyo committed:
Add OnnxConfig for ConvBERT (#16859)

* add OnnxConfig for ConvBert

Co-authored-by: ChainYo <t.chaigneau.tc@gmail.com>

gante committed:
TF: XLA repetition penalty (#16879)

cavdard committed:
Changes in create_optimizer to support tensor parallelism with SMP (#16880)

* changes in create optimizer to support tensor parallelism with SMP

* Update src/transformers/trainer.py

Convert if check to one line.

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Cavdar <dcavdar@a07817b12d7e.ant.amazon.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

patrickvonplaten committed:
[DocTests] Fix some doc tests (#16889)

* [DocTests] Fix some doc tests

* hacky fix

* correct

chainyo committed:
add bigbird typo fixes (#16897)

Co-authored-by: ChainYo <t.chaigneau.tc@gmail.com>

patrickvonplaten committed:
Fix doc test quicktour dataset (#16929)

* fix doc test

* fix doc test

Co-authored-by: Patrick <patrick@pop-os.localdomain>

ydshieh committed:
Add missing ckpt in config docs (#16900)

* add missing ckpt in config docs

* add more missing ckpt in config docs

* fix wrong ckpts

* fix realm ckpt

* fix s2t2

* fix xlm_roberta ckpt

* Fix for deberta v2

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* use only one checkpoint for DPR

* Apply suggestions from code review

Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: NielsRogge <48327001+NielsRogge@users.noreply.github.com>

ydshieh committed:
Fix PyTorch RAG tests GPU OOM (#16881)

* add torch.cuda.empty_cache in some PT RAG tests

* torch.cuda.empty_cache in tearDownModule()

* tearDown()

* add gc.collect()

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ydshieh committed:
Fix RemBertTokenizerFast (#16933)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

gante committed:
TF: XLA logits processors - minimum length, forced eos, and forced bos (#16912)

* XLA min len, forced eos, and forced bos

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

gante committed:
TF: XLA Logits Warpers (#16899)

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

0xrushi committed:
added deit onnx config (#16887)

* added deit onnx config

gante committed:
TF: XLA stable softmax (#16892)


Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

sanchit-gandhi committed:
Replace deprecated logger.warn with warning (#16876)

code-review-doctor committed:
Fix issue probably-meant-fstring found at https://codereview.doctor (#16913)

sgugger committed:
Limit the use of PreTrainedModel.device (#16935)

* Limit the use of PreTrainedModel.device

* Fix

ManuelFay committed:
apply torch int div to layoutlmv2 (#15457)

* apply torch int div

* black linting fixup

* update path to torch_int_div

* clarify imports

agemagician committed:
FIx Iterations for decoder (#16934)

FIx Iterations for decoder

Commit:
Add onnx config for RoFormer (#16861)

* add roformer onnx config

mingboiz committed:
documentation: some minor clean up (#16850)

ftnext committed:
Fix RuntimeError message format (#16906)

tricktreat committed:
use original loaded keys to find mismatched keys (#16920)

anton-l committed:
[Research] Speed up evaluation for XTREME-S (#16785)

* Avoid repeated per-lang filtering

* Language groups and logits preprocessing

* Style

NielsRogge committed:
Add semantic script, trainer (#16834)

* Add first draft

* Improve script and README

* Improve README

* Apply suggestions from code review

* Improve script, add link to resulting model

* Add corresponding test

* Adjust learning rate

ydshieh committed:
Fix HubertRobustTest PT/TF equivalence test on GPU (#16943)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

searchivarius committed:
Misc. fixes for Pytorch QA examples: (#16958)

1. Fixes evaluation errors popping up when you train/eval on squad v2 (one was newly encountered and one that was previously reported Running SQuAD 1.0 sample command raises IndexError #15401 but not completely fixed).
2. Removes boolean arguments that don't use store_true. Please, don't use these: *ANY non-empty string is being converted to True in this case and this clearly is not the desired behavior (and it creates a LOT of confusion).
3. All no-trainer test scripts are now saving metric values in the same way (with the right prefix eval_), which is consistent with the trainer-based versions.
4. Adds forgotten model.eval() in the no-trainer versions. This improved some results, but not everything (see the discussion in the end). Please, see the F1 scores and the discussion below.

NielsRogge committed:
[HF Argparser] Fix parsing of optional boolean arguments (#16946)

* Add fix

* Apply suggestion from code review

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

Yard1 committed:
Fix `distributed_concat` with scalar tensor (#16963)

* Fix `distributed_concat` with scalar tensor

* Update trainer_pt_utils.py

mishig25 committed:
Update custom_models.mdx (#16964)

BertModelForSequenceClassification -> BertForSequenceClassification

sgugger committed:
Fix add-new-model-like when model doesn't support all frameworks (#16966)

sgugger committed:
Fix multiple deletions of the same files in save_pretrained (#16947)

* Fix multiple deletions of the same files in save_pretrained

* Add is_main_process argument

muellerzr committed:
Fixup no_trainer save logic (#16968)

* Fixup all examples

sgugger committed:
Fix doc notebooks links (#16969)

* Fix doc notebooks links

* Remove missing section

ydshieh committed:
Fix check_all_models_are_tested (#16970)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ydshieh committed:
Add -e flag to some GH workflow yml files (#16959)

* Add -e flag

* add check

* create new keys

* run python setup.py build install

* add comments

* change to develop

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

datquocnguyen committed:
Update tokenization_bertweet.py (#16941)

The emoji version must be either 0.5.4 or 0.6.0. Newer emoji versions have been updated to newer versions of the Emoji Charts, thus not consistent with the one used for pre-processing the pre-training Tweet corpus (i.e. not consistent with the vocab).

ydshieh committed:
Update check_models_are_tested to deal with Windows path (#16973)

* fix

* Apply suggestions from code review

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

conan1024hao committed:
Add parameter --config_overrides for run_mlm_wwm.py (#16961)

* dd parameter --config_overrides for run_mlm_wwm.py

* linter

amyeroberts committed:
Rename a class to reflect framework pattern AutoModelXxx -> TFAutoModelXxx (#16993)

ydshieh committed:
set eos_token_id to None to generate until max length (#16989)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

muellerzr committed:
Fix savedir for by epoch (#16996)

sgugger committed:
Update README to latest release (#16997)

ydshieh committed:
use scale=1.0 in floats_tensor called in speech model testers (#17007)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

muellerzr committed:
Update all require decorators to use skipUnless when possible (#16999)

gante committed:
TF: XLA bad words logits processor and list of processors (#16974)

pbelevich committed:
Make create_extended_attention_mask_for_decoder static method (#16893)

tarzanwill committed:
Update README_zh-hans.md (#16977)

Narsil committed:
Updating variable names. (#16445)

Narsil committed:
Revert "Updating variable names. (#16445)" (#17011)

This reverts commit 4f3a14e3c235c8b6b8cd2f5bc448a0cffacddf61.

sgugger committed:
Replace dict/BatchEncoding instance checks by Mapping (#17014)

* Replace dict/BatchEncoding instance checks by Mapping

* Typo

sgugger committed:
Result of new doc style with fixes (#17015)

* Result of new doc style with fixes

* Add last two files

* Bump hf-doc-builder

ydshieh committed:
Add a check on config classes docstring checkpoints (#17012)

* Add the check

* add missing ckpts

* add a list to ignore

* call the added check script

* better regex pattern

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

omarespejel committed:
Add translating guide (#17004)

* Add translating guide

manandey committed:
update docs of length_penalty (#17022)

sanchit-gandhi committed:
[FlaxGenerate] Fix bug in decoder_start_token_id (#17035)

michaelbenayoun committed:
Fx with meta (#16836)

* Add meta proxy

* Uses meta data to trace data dependent control-flow

* Remove commented class

* Handles torch creating functions

* Added type annotation to fix tracing

* Tracing works for everything but T5 and GPT-J

* Almost all previously supported models pass

* All architectures can be traced except T5

* Intermediate commit to have a trace of the comparison operators for HFProxy

* Everything works, except loss computation

* Everything works

* Removed unused import

* Overriden methods do not use underlying ops (linear and torch.matmul), and model attributes are copied to the traced version

* Fix torch_matmul_override

* Change attributes reference to deepcopy

* Remove breakpoint and add torch_index_override

* Small fix

* Fix typo

* Replace asserts by explicit exceptions

sgugger committed:
Fix style

sanchit-gandhi committed:
[Flax(Speech)EncoderDecoder] Fix bug in `decoder_module` (#17036)

* [FlaxSpeechEncoderDecoder] Fix bug in `decoder_module`

* [FlaxEncoderDecoder] Fix bug in `decoder_module`

mpoemsl committed:
Fix typo in RetriBERT docstring (#17018)

JunnYu committed:
add torch.no_grad when in eval mode (#17020)

* add torch.no_grad when in eval mode

* make style quality

sgugger committed:
Disable Flax GPU tests on push (#17042)

NielsRogge committed:
Clean up vision tests (#17024)

* Clean up tests

* Make fixup

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

calpt committed:
[Trainer] Move logic for checkpoint loading into separate methods for easy overriding (#17043)

muellerzr committed:
Update no_trainer examples to use new logger (#17044)

* Propagate and fix imports

muellerzr committed:
Fix no_trainer examples to properly calculate the number of samples (#17046)

* Update all examples to properly calculate progress bar

NielsRogge committed:
Add YOLOS (#16848)

* First draft

* Add YolosForObjectDetection

* Make forward pass work

* Add mid position embeddings

* Add interpolation of position encodings

* Add expected values

* Add YOLOS to tests

* Add integration test

* Support tiny model as well

* Support all models in conversion script

* Remove mid_pe_size attribute

* Make more tests pass

* Add model to README and fix config

* Add copied from statements

* Rename base_model_prefix to vit

* Add missing YOLOS_PRETRAINED_CONFIG_ARCHIVE_MAP

* Apply suggestions from code review

* Apply more suggestions from code review

* Convert remaining checkpoints

* Improve docstrings

* Add YolosFeatureExtractor

* Add feature extractor to docs

* Add corresponding tests

* Fix style

* Fix docs

* Apply suggestion from code review

* Fix bad rebase

* Fix some more bad rebase

* Fix missing character

* Improve docs and variable names

Co-authored-by: Niels Rogge <nielsrogge@Nielss-MacBook-Pro.local>

LysandreJik committed:
Allow all imports from transformers (#17050)

LysandreJik committed:
Make the sacremoses dependency optional (#17049)

* Make sacremoses optional

* Pickle

sgugger committed:
Clean up setup.py (#17045)

* Clean up setup.py

* Trigger CI

* Upgrade Python used

patrickvonplaten committed:
[T5 Tokenizer] Model has no fixed position ids - there is no hardcode… (#16990)

* [T5 Tokenizer] Model has no fixed position ids - there is no hardcoded max length

* [T5 Tokenizer] Model has no fixed position ids - there is no hardcoded max length

* correct t5 tokenizer

* correct t5 tokenizer

* fix test

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* finish

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

sanchit-gandhi committed:
[FlaxBert] Add ForCausalLM (#16995)

* [FlaxBert] Add ForCausalLM

* make style

* fix output attentions

* Add RobertaForCausalLM

* remove comment

* fix fx-to-pt model loading

* remove comment

* add modeling tests

* add enc-dec model tests

* add big_bird

* add electra

* make style

* make repo-consitency

* add to docs

* remove roberta test

* quality

* amend cookiecutter

* fix attention_mask bug in flax bert model tester

* tighten pt-fx thresholds to 1e-5

* add 'copied from' statements

* amend 'copied from' statements

* amend 'copied from' statements

* quality

ydshieh committed:
Move test model folders (#17034)

* move test model folders (TODO: fix imports and others)

* fix (potentially partially) imports (in model test modules)

* fix (potentially partially) imports (in tokenization test modules)

* fix (potentially partially) imports (in feature extraction test modules)

* fix import utils.test_modeling_tf_core

* fix path ../fixtures/

* fix imports about generation.test_generation_flax_utils

* fix more imports

* fix fixture path

* fix get_test_dir

* update module_to_test_file

* fix get_tests_dir from wrong transformers.utils

* update config.yml (CircleCI)

* fix style

* remove missing imports

* update new model script

* update check_repo

* update SPECIAL_MODULE_TO_TEST_MAP

* fix style

* add __init__

* update self-scheduled

* fix add_new_model scripts

* check one way to get location back

* python setup.py build install

* fix import in test auto

* update self-scheduled.yml

* update slack notification script

* Add comments about artifact names

* fix for yolos

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

sgugger committed:
Make Trainer compatible with sharded checkpoints (#17053)

* Make Trainer compatible with sharded checkpoints

* Add doc

sgugger committed:
Remove Python and use v2 action (#17059)

sgugger committed:
Fix RNG reload in resume training from epoch checkpoint (#17055)

* Fix RNG reload in resume training from epoch checkpoint

* Fix test

sgugger committed:
Remove fetch in model templates test

pbelevich committed:
Remove device parameter from create_extended_attention_mask_for_decoder (#16894)

thomasw21 committed:
Fix hashing for deduplication (#17048)

lewtun committed:
Skip RoFormer ONNX test if rjieba not installed (#16981)

* Skip RoFormer ONNX test if rjieba not installed

* Update deps table

* Skip RoFormer serialization test

* Fix RoFormer vocab

* Add rjieba to CircleCI

lewtun committed:
Remove masked image modeling from BEIT ONNX export (#16980)

* Add masked image modelling to task mapping

* Refactor ONNX features to be listed alphabetically

* Add warning about BEiT masked image modeling

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

sgugger committed:
Make sure telemetry arguments are not returned as unused kwargs (#17063)

* Make sure telemetry arguments are not returned as unused kwargs

* Fix test

sayakpaul committed:
Add Data2Vec for Vision in TF (#17008)

* add utilities till TFData2VecVisionLayer.

* chore: pass window_size to attention layer.

* feat: add TFData2VecVisionRelativePositionBias.

* feat: initial implementation ready for tf data2vec.

* fix: relative position bias index, table to be fixed.

* chore: implementation added, tests remaining.

* add: tests, other PR files.

* fix: code quality.

* fix: import structure in init.

* chore: run make fix-copies.

* chore: address PR feedback (round I).

* chore: styling nit.

* fix: tests due to removal of to_2tuple().

* chore: rebase with upstream main and move the test.

* Update src/transformers/models/auto/modeling_tf_auto.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update src/transformers/models/auto/modeling_tf_auto.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* fix: layer call.

* chore: remove from_pt=True and rerun test.

* chore: remove cast and tf.divide.

* chore: minor edits to the test script.

* Update src/transformers/models/data2vec/modeling_tf_data2vec_vision.py

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

* fix: expand() on TF tensors with broadcast_to().

* fix: test import.

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

dependabot[bot] committed:
Bump notebook in /examples/research_projects/visual_bert (#16635)

Bumps [notebook](http://jupyter.org) from 6.4.1 to 6.4.10.

---
updated-dependencies:
- dependency-name: notebook
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump notebook from 6.4.1 to 6.4.10 in /examples/research_projects/lxmert (#16634)

Bumps [notebook](http://jupyter.org) from 6.4.1 to 6.4.10.

---
updated-dependencies:
- dependency-name: notebook
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

karthikrangasai committed:
Type hint complete Albert model file. (#16682)

* Type hint complete Albert model file.

* Update typing.

* Update src/transformers/models/albert/modeling_albert.py

Co-authored-by: Matt <Rocketknight1@users.noreply.github.com>

sgugger committed:
Deprecate model templates (#17062)

* Deprecate model templates

* Address review comments

muellerzr committed:
Update to build via git for accelerate (#17084)

seanmor5 committed:
Allow saved_model export of TFCLIPModel in save_pretrained (#16886)

* CLIP Serving

* Add type hints per code review

* Use black, flake8, and isort

* Update src/transformers/models/clip/modeling_tf_clip.py

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>

* Rollback serving_output and add TODO

* Remove irrelevant portions of failing tests

* Revert "Rollback serving_output and add TODO"

This reverts commit a4abfa6ba3b7875a13538dbc2ddc4eb17dfcca8d.

* Rollback to original test/serving_output

* Fix unused var

* Apply suggestions from code review

* Update formatting with black

* Fix style again from rebase

* Update tests/models/clip/test_modeling_tf_clip.py

Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

Co-authored-by: Joao Gante <joaofranciscocardosogante@gmail.com>
Co-authored-by: Sean Moriarity <sean.l.moriarity.mil@army.mil>
Co-authored-by: Yih-Dar <2521628+ydshieh@users.noreply.github.com>

deutschmn committed:
Fix DeBERTa `token_type_ids` (#17082)

stevhliu committed:
📝 open fresh PR for pipeline doctests (#17073)

ydshieh committed:
minor change on TF Data2Vec test (#17085)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

duedme committed:
Added spanish translation of autoclass_tutorial. (#17069)

* Added spanish translation of autoclass_tutorial.
Added 'local' and 'title' fields for autoclass_tutorial.

* Fixed autoclass_tutorial title in _toctree.yml and autoclass_tutorial.mdx

robotjellyzone committed:
type hints for pytorch models (#17064)

* type hints for pytorch models

* fixed import error

* fixed some errors

robsmith155 committed:
Add type hints for BERTGeneration (#17047)

Added type hints for the BERTGenerationEncoder and BERTGenerationDecoder
classes.

orieg committed:
Fix MLflowCallback and add support for MLFLOW_EXPERIMENT_NAME (#17091)

* Fix use of mlflow.active_run() and add proper support for MLFLOW_EXPERIMENT_NAME

* Fix code style (make style)

sgugger committed:
Remove torchhub test (#17097)

ydshieh committed:
fix missing "models" in pipeline test module (#17090)

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

stevhliu committed:
Fix link to example scripts (#17103)

ydshieh committed:
Fix self-push CI report path in cat (#17111)

* fix report cat path

* fix report cat path

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

nandwalritik committed:
Added BigBirdPegasus onnx config (#17104)

* Add onnx configuration for bigbird-pegasus

* Modify docs

AnugunjNaman committed:
Update remove num_stages

AnugunjNaman committed:
Update remove num_stages

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_16575). All of your documentation changes will be reflected on that endpoint.

AnugunjNaman committed:
Update

AnugunjNaman committed:
Update fix-copies

AnugunjNaman committed:
Update fix-copies

AnugunjNaman commented:
@NielsRogge can you review it and suggest any further changes.
I'm not sure how you want to modify `cls_token` part. Since modifying in `CvtStage` section (stopping the split) will change shape of hidden states (stored in all hidden states and passing of 4D shape for CNN in next layer). I leave that part to you on how you want to change it to pass it over different classes further.

I have run the `make fix-copies` and done docstrings part. I think everything is done apart from change you wanted to make for `cls_token`.

P.S. Can we sit for an hour and so to finish this PR. I can then finally move on from it. I know you're busy but I hope you can find some time.

NielsRogge commented:
Hi,

It seems that there was an issue with rebasing as the "files changed" tab now shows that 800+ files have changed.

Could you open a new, clean PR?

AnugunjNaman commented:
@NielsRogge How should I do that. Create a new branch and copy all the codes?
